File: example/python/yaml-to-json.py
#!/usr/bin/env python3
"""Run with: python yaml-to-json.py [input.yaml]"""

import sys
import json

# Add parent python/lib to path for development
sys.path.insert(0, '../../python/lib')

import yamlstar

def main(args):
    yaml_file = args[0] if args else "sample.yaml"
    print(f"YAMLStar Example - Loading {yaml_file} and outputting JSON\n")

    with open(yaml_file) as f:
        yaml_content = f.read()

    print("Input YAML:")
    print(yaml_content)
    print("\n---\n")

    ys = yamlstar.YAMLStar()
    data = ys.load(yaml_content)

    json_output = json.dumps(data, indent=2)

    print("Output JSON:")
    print(json_output)

if __name__ == '__main__':
    main(sys.argv[1:])

File: example/go/yaml-to-json.go
package main

import (
	"encoding/json"
	"fmt"
	"os"

	"github.com/yaml/yamlstar"
)

func main() {
	yamlFile := "../sample.yaml"
	if len(os.Args) > 1 {
		yamlFile = os.Args[1]
	}

	fmt.Printf("YAMLStar Example - Loading %s and outputting JSON\n\n", yamlFile)

	yamlContent, err := os.ReadFile(yamlFile)
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error reading file: %v\n", err)
		os.Exit(1)
	}

	fmt.Println("Input YAML:")
	fmt.Println(string(yamlContent))
	fmt.Print("\n---\n\n")

	data, err := yamlstar.Load(string(yamlContent))
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error parsing YAML: %v\n", err)
		os.Exit(1)
	}

	jsonOutput, err := json.MarshalIndent(data, "", "  ")
	if err != nil {
		fmt.Fprintf(os.Stderr, "Error encoding JSON: %v\n", err)
		os.Exit(1)
	}

	fmt.Println("Output JSON:")
	fmt.Println(string(jsonOutput))
}
